proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,315 @@
+an open issue that has been discussed by the maintainers or the community. They
+often end up lingering with no communication from the maintainers to indicate
+the feasibility or appropriateness of the changes. As an example, [`aws-cli`
+pull request 2636](https://github.com/aws/aws-cli/pull/2636)  adds functionality
+to configure MFA. It adds or alters 231 lines of code across five files, and is
+a significant new feature. There is no open issue or discussion about the
+feature other than on the PR. It has no comments from AWS employees, other than
+to indicate that it is a “large” pull request, since 2017. Conversation
+continued for a year on the PR by interested users with no other interaction or
+guidance from AWS.
+
+A idea that has not been vetted by the community and the maintainers is
+difficult to approve; it's hard to have a conversation when it involves an idea
+proposal as well as code changes. It usually ends in an unmerged pull request or
+significant code revision, which is effort that was unnecessarily expended by
+the user and potentially the maintainer. We need a structured process for the
+life cycle of an idea that can culminate in a community contribution. We also
+need to prioritize issues against other requests from the community given the
+effort to contribute for users and review for maintainers.
+
+## Specification
+
+This section defines:
+
+* [Current process](#current-process)
+* [Proposed improvements](#proposed-improvements)
+
+### Current Process
+
+Customer feature requests are reviewed for general suitability and uniqueness,
+but are not actively commented on by maintainers to move the feature forward to
+an implementation. Pull requests are reviewed in an *ad hoc* fashion by various
+members of the maintainer team, but similarly are not actively processed to
+continue towards acceptance. This has contributed to the current status of 189
+pull requests open in the AWS CLI GitHub repository.
+
+### Proposed Improvements
+ 
+Maintainers that actively engage with the community contributors to identify
+suitable implementations for requested features and provide gudiance during the
+process will result in a higher percentage of successful community
+contributions. In addition, we will publish a contributing guide that describes
+the life cycle of a feature request that can culminate in a code contribution
+from the community through a GitHub pull request. The guide describes the triage
+and review of open feature requests with the goal of identifying issues where
+community contributions would be welcome. It defines what is required to move a
+feature request to be available for community contribution. It documents the
+review process and sets community expectations around communications to move a
+contribution forward. Figure A demonstrates a high level view of the proposed
+process.
+
+The maintainers will actively review GitHub issues for feature requests to
```

Comment:
Does this proposal include how we are going to approach this? I think we would need some kind of weekly meeting or time carved out for this specifically if we are going to be able to keep it up.